core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -639,72 +708,26 @@ int getMaxItems() {
-        // the old value reference is no longer there so we have nothing to do. Note that in case of non-ZC removal the
-        // loop in remove() will lookup the key again so we will still return the previous value
-        if (operation == Operation.REMOVE) {
-            return true; // this is a remove, no need to try again and return doesn't matter
-        }
-
-        // the operation is either NO_OP, PUT, PUT_IF_ABSENT, COMPUTE
-        long foundValueReference = getValueReference(opData.entryIndex);
-
-        if (opData.newValueReference == foundValueReference) {
-            return true; // someone helped
-        } else if (foundValueReference == INVALID_VALUE_REFERENCE) {
-            // the value reference was deleted, retry the attach
-            opData.oldValueReference = INVALID_VALUE_REFERENCE;
-            return pointToValue(opData); // remove completed, try again
-        } else if (operation == Operation.PUT_IF_ABSENT) {
-            return false; // too late
-        } else if (operation == Operation.COMPUTE) {
-            Slice valueSlice = buildValueSlice(foundValueReference);
-            ValueUtils.ValueResult succ = ValueUtils.compute(valueSlice, opData.computer);
-
-            if (succ != ValueUtils.ValueResult.SUCCESS) {
-                // we tried to perform the compute but the value was deleted,
-                // we can get to pointToValue with Operation.COMPUTE only from PIACIP
-                // retry to make a put and to attach the new value
-                opData.oldValueReference = foundValueReference;
-                return pointToValue(opData);
-            }
-        }
-        // This is a put, in which case operation should restart,
-        // or PIACIP compute happened, which should return false
-        return false;
-    }
-
-    /**
-     * used by put/putIfAbsent/remove and rebalancer
-     */
-    private boolean pointToValueCAS(OpData opData) {
-        if (casEntriesArrayLong(opData.entryIndex, OFFSET.VALUE_REFERENCE, opData.oldValueReference,
+    NovaValueUtils.NovaResult linkValue(OpData opData) {
+        if (!casEntriesArrayLong(opData.entryIndex, OFFSET.VALUE_REFERENCE, opData.oldValueReference,
                 opData.newValueReference)) {
-            // update statistics only by thread that CASed
-            if (opData.oldValueReference == INVALID_VALUE_REFERENCE && opData.newValueReference != INVALID_VALUE_REFERENCE) {
-                // previously a remove
-                statistics.incrementAddedCount();
-                externalSize.incrementAndGet();
-            } else if (opData.oldValueReference != INVALID_VALUE_REFERENCE && opData.newValueReference == INVALID_VALUE_REFERENCE) {
-                // removing
-                statistics.decrementAddedCount();
-                externalSize.decrementAndGet();
-            }
-            return true;
+            return FALSE;
```

Comment:
How does it returns FALSE? Is NovaValueUtils.NovaResult an enum?

Reply:
Yes